include/LeMonADE/analyzer/AnalyzerRadiusOfGyration.h

```diff
@@ -59,227 +44,229 @@ along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
 	
+private:
+	//! typedef for the underlying container holding the monomers
+	typedef typename IngredientsType::molecules_type molecules_type;
+	//! reference to the complete system
 	const IngredientsType& ingredients;
-	
+	//! Rg2 is calculated for the groups in this vector
 	std::vector<MonomerGroup<molecules_type> > groups;
+	//! timeseries of the Rg^2. Components: [0]-> Rg^2_x, [1]->Rg^2_y, [2]->Rg^2_z, [3]:Rg^2_tot
+	std::vector< std::vector<double> > Rg2TimeSeries;
+	//! vector of mcs times for writing the time series
+	std::vector<double> MCSTimes;
+	//! max length of internal buffer for each coordinate, before saving to disk
+	uint32_t bufferSize;
+	//! name of output files are outputFilePrefix_averages.dat and outputFilePrefix_timeseries.dat
+	std::string outputFile;
+	//! flag used in dumping time series output
+	bool isFirstFileDump;
+	//! save the current values in Rg2TimeSeriesX, etc., to disk
+	void dumpTimeSeries();
+	//! calculate the Rg squared of the monomer group
+	VectorDouble3 calculateRg2Components(const MonomerGroup<molecules_type>& group) const;
 	
-	TimeSeries<VectorDouble3>  Rg2TimeSeries;
-	std::vector<Moments<double> > Rg_squaredX,Rg_squaredY,Rg_squaredZ,Rg_squared;
-	
-	//! mcs below this number are not considered in analysis
-	uint64_t lowerLimit;
-	
-	//! prefix for output filename
-	std::string outputFilePrefix;
 public:
   
-	/*
-	 * there are three different constructors: 
-	 * - the first calculates the Rg^2 of the complete system
-	 * - the second takes a monomer group as additional argument and 
-	 *   calculates the average Rg^2 of the monomers in this group
-	 * - the third takes a std::vector of monomer groups and calculates
-	 *   the Rg^2 of all molecules in this group. The time series output is
-	 *   then the average Rg^2, averaged over both time and groups.
-	 * */
-	AnalyzerRadiusOfGyration(const IngredientsType& ing,std::string filenamePrefix="",uint64_t lowerMcsLimit=0);
-	AnalyzerRadiusOfGyration(const IngredientsType& ing,
-				 const MonomerGroup<molecules_type>& g,
-			  std::string filenamePrefix="",
-			  uint64_t lowerMcsLimit=0);
+	//! constructor
 	AnalyzerRadiusOfGyration(const IngredientsType& ing,
-				 const std::vector<MonomerGroup<molecules_type> >& g,
-			  std::string filenamePrefix="",
-			  uint64_t lowerMcsLimit=0);
+			  std::string filename="Rg2TimeSeries.dat");
 	
+	//! destructor. does nothing
 	virtual ~AnalyzerRadiusOfGyration(){}
-	
+	//! Initializes data structures. Called by TaskManager::initialize()
+	virtual void initialize();
+	//! Calculates the Rg2 for the current timestep. Called by TaskManager::execute()
 	virtual bool execute();
+	//! Writes the final results to file
 	virtual void cleanup();
+	//! Set the number of values, after which the time series is saved to disk
+	void setBufferSize(uint32_t size){bufferSize=size;}
+	//! Set the groups to be analyzed
+	void setMonomerGroups(std::vector<MonomerGroup<molecules_type> > groupVector){groups=groupVector;}
```

Comment:
Make this function protected to show it is designed to use it in a derived class.